File: src/main/java/fit/wenchao/db/JavaClassName.java
package fit.wenchao.db;

import lombok.Data;

@Data
public class JavaClassName {

    String name;

    JavaPackage javaPackage;

    public static JavaClassName fromLowerUnderScore(JavaPackage javaPackage, String name, String prefix) {
        JavaClassName javaClassName = new JavaClassName();
        javaClassName.javaPackage = javaPackage;
        String underScore = VarCaseConvertUtils.lowerUnderScore2LowerCamel(name);
        if (underScore.length() >= 1) {
            underScore = ("" + underScore.charAt(0)).toUpperCase() + underScore.substring(1);
        }
        javaClassName.setName(underScore + prefix);
        return javaClassName;
    }

    public String getFullName() {
        return javaPackage.getDotSplitName() + "." + name;
    }

    public String toJavaSrcFileName() {
        return name + ".java";
    }

    public String toString() {
        return name;
    }
}

File: src/main/kotlin/fit/wenchao/db/JavaClassName.kt
package fit.wenchao.db

import fit.wenchao.db.constants.Lang

class JavaClassName(var name: String, var javaPackage: JavaPackage) {

    companion object {
        fun fromLowerUnderScore(javaPackage: JavaPackage, name: String, prefix: String): JavaClassName {
            val underScore = VarCaseConvertUtils.lowerUnderScore2LowerCamel(name)
            val capitalized = underScore.replaceFirstChar { it.uppercase() }
            return JavaClassName(capitalized + prefix, javaPackage)
        }
    }

    fun getFullName(): String {
        return "${javaPackage.dotSplitName}.$name"
    }

    fun toJavaSrcFileName(): String {
        return "$name.java"
    }

    fun toKotlinSrcFileName(): String {
        return "$name.kt"
    }

    fun toSrcFileName(lang: Lang): String {
        return if (lang == Lang.KOTLIN) {
            toKotlinSrcFileName()
        } else {
            toJavaSrcFileName()
        }
    }

    override fun toString(): String {
        return name
    }
}
